playbook/controllers.go

```diff
@@ -143,6 +144,22 @@ func HandlePlaybookRun(c echo.Context) error {
 		return c.JSON(http.StatusInternalServerError, api.HTTPError{Error: err.Error(), Message: "failed to create playbook run"})
 	}
 
+	go func() {
+		// Create a new context because the original request context will be cancelled immediately
+		// after exiting out of this handler.
+		ctx, cancel := api.NewContext(ctx.DB(), ctx.Pool()).WithUser(ctx.User()).WithTimeout(time.Second * 20)
```

Comment:
This type of new decentralised context creation / cloning  I want to avoid - If we need a context that lives past the request then it should be a new context function that passes in relevant details
 e.g. 

For example we wanted to enable tracing by setting a trace header or request, it would be done in the api.NewContextFromHttpRequest(ctx) or ctx.NewBackgroundContext()

   or` api.NewContextFromHttpRequest() { api.DefaultContext.WithTrace(req.headers['trace'] = 'true').WithValue("startedBy", user.id|)`

Reply:
I've added a new `WithDetachedTimeout` to the context which essentially preserves everything in the context apart from the timeout.

https://github.com/flanksource/incident-commander/blob/eaacae2e2b3d98fe5974e0626579e15a9ff9ca70/api/context_test.go

Reply:
It uses a new function from the context package introduced in go 1.21